src/zenml/stack/stack_component.py

```diff
@@ -109,6 +95,71 @@ def __init__(self, **kwargs: Any) -> None:
 
         super().__init__(**kwargs)
 
+
+class StackComponent:
+    """Abstract StackComponent class for all components of a ZenML stack."""
+
+    def __init__(
+        self,
+        name: str,
+        id: UUID,
+        config: StackComponentConfig,
+        flavor: str,
+        type: StackComponentType,
+        user: UUID,
+        project: UUID,
+        *args,
+        **kwargs,
+    ):
+        self.id = id
+        self.name = name
+        self.config = config
+        self.flavor = flavor
+        self.type = type
+        self.user = user
+        self.project = project
+
+    @classmethod
+    def from_model(cls, component_model: "ComponentModel") -> "StackComponent":
+        from zenml.repository import Repository
+
+        flavor_model = Repository(
+            skip_repository_check=True  # noqa
+        ).get_flavor_by_name_and_type(
+            name=component_model.flavor,
+            component_type=component_model.type,
+        )
+
+        try:
+            from zenml.stack import Flavor
+
+            flavor = Flavor.from_model(flavor_model)
+        except (ModuleNotFoundError, ImportError, NotImplementedError):
+            raise ImportError(f"tmp")
+
+        configuration = flavor.config_class(**component_model.configuration)
+
+        return flavor.implementation_class(
+            user=component_model.user,
+            project=component_model.project,
+            name=component_model.name,
+            id=component_model.id,
+            config=configuration,
+            flavor=component_model.flavor,
+            type=component_model.type,
+        )
+
+    def to_model(self) -> "ComponentModel":
+        return ComponentModel(
+            user=self.user,
+            project=self.project,
+            id=self.id,
+            type=self.type,
+            flavor_name=self.flavor,
+            name=self.name,
+            configuration=self.config.dict(),
```

Comment:
I'm not really sure here, but the default factory of the created field of the ComponentModel will be triggered here and a false creation date for the component will be generated. We have to make sure this isn't falsely stored or reported to the user. Alternativley it could be queried from the zen_store using the id